### DISCUSSION TITLE
Base depth in AGS files not always per specification

### DISCUSSION THREAD

**Discussion by hardreddata** [australiangeomechanics/agsformat#33] [Q&A] [answered]
Hopefully outsiders are welcome here.

This came up recently.

To my reading the specification suggests that both Top and Base depths are required in GEOL and DREM and DETL tables. The Base is part of the table key.

In practice we often encounter AGS files (technically speaking to AGS 3.1 RTA 1.1 here) where DREM and DETL do not always contain a Base depth.

Any advice or guidance is very welcome.

**Comment by SolidRockProgrammer** [comment, marked as answer]
Unless there is an R in Status, data is not required.
A key heading must be present in a group, but can be empty.

**Reply by hardreddata** [reply]
Thanks @SolidRockProgrammer for the prompt response. 

I didn't see the R and *R.

There is some interesting advice in the spreadsheet. I appreciate the way the table is keyed and requirements for allowing null fields can be distinct from how users may behave.

![image](https://user-images.githubusercontent.com/42530666/149241191-c9e88c6a-2985-45f6-87ed-b894a24e9422.png)
